pyinfra/api/connectors/winrm.py

```diff
@@ -0,0 +1,213 @@
+
+    Args:
+        state (``pyinfra.api.State`` obj): state object for this command
+        hostname (string): hostname of the target
+        command (string): actual command to execute
+        get_pty (boolean): Not used for WINRM
+        sudo (boolean): Not used for WINRM
+        sudo_user (string): Not used for WINRM
+        use_sudo_login(boolean): Not used for WINRM
+        use_su_login(boolean): Not used for WINRM
+        preserve_sudo_env(boolean): Not used for WINRM
+    TODO: check if winrm has a timeout and use timeout param
+        timeout (int): timeout for this command to complete before erroring
+        stdin (string): Not used for WINRM
+        success_exit_codes (list): all values in the list that will return success
+        print_output (boolean): print the output
+        TODO print_intput (boolean): print the input
+        return_combined_output (boolean): combine the stdout and stderr lists
+        shell_executable (string): shell to use - 'sh'=cmd, 'ps'=powershell(default)
+        env (dict): envrionment variables to set
+
+    Returns:
+        tuple: (exit_code, stdout, stderr)
+        stdout and stderr are both lists of strings from each buffer.
+    '''
+
+    command = make_win_command(command, **command_kwargs)
+
+    logger.debug('Running command on {0}: {1}'.format(host.name, command))
+
+    # get rid of leading/trailing quote
+    tmp_command = command.strip("'")
+
+    if print_output:
+        click.echo('{0}>>> {1}'.format(host.print_prefix, command))
+
+    if not shell_executable:
+        shell_executable = 'ps'
+    logger.debug('shell_executable:{0}'.format(shell_executable))
+
+    # default windows to use ps, but allow it to be overridden
+    if shell_executable in ['cmd']:
+        response = host.connection.run_cmd(tmp_command)
+    else:
+        response = host.connection.run_ps(tmp_command)
+
+    return_code = response.status_code
+    logger.debug('response:{}'.format(response))
+
+    std_out_str = response.std_out.decode('utf-8')
+    std_err_str = response.std_err.decode('utf-8')
+
+    # split on '\r\n' (windows newlines)
+    std_out = std_out_str.split('\r\n')
+    std_err = std_err_str.split('\r\n')
+
+    logger.debug('std_out:{}'.format(std_out))
+    logger.debug('std_err:{}'.format(std_err))
+
+    if print_output:
+        click.echo('{0}>>> {1}'.format(host.print_prefix, '\n'.join(std_out)))
+
+    if success_exit_codes:
+        status = return_code in success_exit_codes
+    else:
+        status = return_code == 0
+
+    logger.debug('Command exit status: {0}'.format(status))
+
+    if return_combined_output:
+        return status, std_out + std_err
+
+    return status, std_out, std_err
+
+
+def get_file(
+    state, host, remote_filename, filename_or_io,
+    **command_kwargs
+):
+    # TODO: implement
+    return True
+
+
+def put_file(
+    state, host, filename_or_io, remote_filename,
+    **command_kwargs
+):
+    # TODO: implement
+    return True
```

Comment:
As above, should `raise NotImplementedError` here.

Reply:
Please review. Not sure if this is what you wanted.

Reply:
Works for me! There's a python builtin `NotImplementedError` but the pyinfra exception works too - probably more sensible in this case!